src/components/structures/ScrollPanel.js

```diff
@@ -77,111 +78,48 @@ if (DEBUG_SCROLL) {
-
-        /* onUnfillRequest(backwards): a callback which is called on scroll when
-         * there are children elements that are far out of view and could be removed
-         * without causing pagination to occur.
-         *
-         * This function should accept a boolean, which is true to indicate the back/top
-         * of the panel and false otherwise, and a scroll token, which refers to the
-         * first element to remove if removing from the front/bottom, and last element
-         * to remove if removing from the back/top.
-         */
-        onUnfillRequest: React.PropTypes.func,
-
-        /* onScroll: a callback which is called whenever any scroll happens.
-         */
-        onScroll: React.PropTypes.func,
-
-        /* onResize: a callback which is called whenever the Gemini scroll
-         * panel is resized
-         */
-        onResize: React.PropTypes.func,
-
-        /* className: classnames to add to the top-level div
-         */
-        className: React.PropTypes.string,
-
-        /* style: styles to add to the top-level div
-         */
-        style: React.PropTypes.object,
-    },
-
-    getDefaultProps: function() {
-        return {
-            stickyBottom: true,
-            startAtBottom: true,
-            onFillRequest: function(backwards) { return Promise.resolve(false); },
-            onUnfillRequest: function(backwards, scrollToken) {},
-            onScroll: function() {},
-        };
-    },
+export default class ScrollPanel extends StickyContainer {
+
+    constructor() {
+        super();
+        this.onResize = this.onResize.bind(this);
+        this.onScroll = this.onScroll.bind(this);
+    }
 
-    componentWillMount: function() {
+    componentWillMount() {
         this._pendingFillRequests = {b: null, f: null};
         this.resetScrollState();
-    },
+    }
 
-    componentDidMount: function() {
+    componentDidMount() {
         this.checkFillState();
-    },
+    }
 
-    componentDidUpdate: function() {
+    componentDidUpdate() {
         // after adding event tiles, we may need to tweak the scroll (either to
         // keep at the bottom of the timeline, or to maintain the view after
         // adding events to the top).
         //
         // This will also re-check the fill state, in case the paginate was inadequate
         this.checkScroll();
-    },
+    }
 
-    componentWillUnmount: function() {
+    componentWillUnmount() {
         // set a boolean to say we've been unmounted, which any pending
         // promises can use to throw away their results.
         //
         // (We could use isMounted(), but facebook have deprecated that.)
         this.unmounted = true;
-    },
+    }
 
-    onScroll: function(ev) {
+    onScroll(ev) {
         var sn = this._getScrollNode();
         debuglog("Scroll event: offset now:", sn.scrollTop,
                  "_lastSetScroll:", this._lastSetScroll);
 
+        this.node = sn;
```

Comment:
unused?

Reply:
Actually no, this is why any of the sticky stuff works and I 100% should have commented as such.